### ISSUE TITLE
Support more storage driver type

### ISSUE THREAD

**Issue by demonwy** [ljfranklin/terraform-resource#40]
Hi @ljfranklin,

   I am intend to use terraform resource type in my concourse pipeline, to create IaaS resources.
   But found it only support S3 as storage driver by now.

   Unfortunately, I need to upload the file to [OSS](https://www.alibabacloud.com/product/oss?spm=a3c0i.7911826.675768.dnavproductsb1.5c62b487mXsmFu#product-details), which is a Cloud Object Storage Service of Alibaba. Do you have plan to support it?

  [OSS API overviews](https://www.alibabacloud.com/help/doc-detail/31948.htm?spm=a3c0i.l31815en.a3.61.2c7c654f6jgpYH)

Best Regards
YueWang

**Comment by ljfranklin** [comment]
Hi @demonwy, thanks for reaching out. In order to support more backends, I'm trying to move away completely from the custom storage drivers to the built-in Terraform backends: https://github.com/ljfranklin/terraform-resource/issues/39. Unfortunately I don't see OSS listed under the [supported terraform backends](https://www.terraform.io/docs/backends/types/index.html). If you'd like Terraform to support OSS as a backend, you'll have to open an issue or PR on [Terraform](https://github.com/hashicorp/terraform).

Since I'm removing the storage drivers in #39, I'm going to close this out as won't fix. But if OSS is ever added as a built-in Terraform backend you'll be able to use it once #39 is finished.

**Comment by demonwy** [comment]
Got it!
Thanks @ljfranklin
